src/PowerShellEditorServices.Host/LanguageServer.cs

```diff
@@ -3,98 +3,76 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 //
 
-using DebugAdapterMessages = Microsoft.PowerShell.EditorServices.Protocol.DebugAdapter;
 using Microsoft.PowerShell.EditorServices.Protocol.LanguageServer;
 using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol;
+using Microsoft.PowerShell.EditorServices.Protocol.MessageProtocol.Channel;
 using Microsoft.PowerShell.EditorServices.Utility;
 using Nito.AsyncEx;
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
 using System.Management.Automation;
 using System.Management.Automation.Language;
 using System.Text.RegularExpressions;
 using System.Threading;
 using System.Threading.Tasks;
-using System.IO;
+using DebugAdapterMessages = Microsoft.PowerShell.EditorServices.Protocol.DebugAdapter;
 
 namespace Microsoft.PowerShell.EditorServices.Host
 {
-    internal class LanguageServer : IMessageProcessor
+    internal class LanguageServer : LanguageServerBase 
```

Comment:
I think moving these classes to Protocol is a good idea, there's no particular reason to make them only accessible through the host.

By the way, the reason why there are *Base implementations for the language/debug servers is that my next big change is factoring out the common protocol implementation pieces into a shared .NET library for speaking the VS Code protocol.  Much of the stuff that's currently in the Protocol library will get moved out to (yet another) library that will get used by other projects.